src/lib/common/sol-mainloop.c

```diff
@@ -104,6 +105,12 @@ sol_init(void)
 
     sol_log_domain_init_level(SOL_LOG_DOMAIN);
 
+#ifdef MODULES
```

Comment:
also, I've asked @ceolin to review the init/shutdowns... he will change it so we always have the calls, but in some cases they are empty. You may choose to take this approach right now and save him some work.